assets/js/data/cart/push-changes.ts

```diff
@@ -100,11 +100,9 @@ const updateCustomerData = debounce( (): void => {
 	}
 
 	if ( Object.keys( customerDataToUpdate ).length ) {
+		removeAllNotices();
```

Comment:
This ensures error notices are removed before trying to update the customer data, in the case the API returns an error we can re-add them.